autosklearn/automl.py

```diff
@@ -1906,6 +1907,9 @@ def show_models(self) -> Dict[int, Any]:
         """
 
         ensemble_dict = {}
+        if self._ensemble_size == 0:
+            self._logger.warning('No models in the ensemble. Kindly check the ensemble size.')
```

Comment:
As this is a front facing API, this warning should be emitted to the user and not just logged silently, in fact there's not much point in logging it silently as it wouldn't help with debugging much.

```python
import warnings
warnings.warn(msg)
```
Also, I don't think `_ensemble_size` is the best proxy for checking this. A use could pass `_ensemble_size = 10` but auto-sklearn failed to find any models, resulting in `self._ensemble_size = 10` but having no `models_` property.

Take a look at [`_load_models()`](https://github.com/automl/auto-sklearn/blob/master/autosklearn/automl.py#L1510) for the logic of `self.ensemble_` and `self.models_`.

While thinking about it, there's probably three separate conditions to check for:
* Has autosklearn been fitted? If not raise a RuntimeError
* Has autosklearn been given a paramter of `ensemble_size=0`? If so, issue a warning and give an empty dict.
* Has autosklearn been given a parameter `ensemble_size > 0` but there is no ensemble to load from? If so, issue a warning and give an empty dict.
* Has 
I think checking `self.ensemble_` for `None` should be good enough

I would change it to
```
# At top
import warnings

if self._ensemble_size == 0
```